pinot-hadoop/src/main/java/org/apache/pinot/hadoop/job/SegmentPreprocessingJob.java

```diff
@@ -505,6 +377,86 @@ private void addHashCodeField(Set<Schema.Field> fieldSet) {
     fieldSet.add(hashCodeField);
   }
 
+  private void setTableConfigAndSchema() throws IOException {
+    // If push locations, table config, and schema are not configured, this does not necessarily mean that segments
+    // cannot be created. We should allow the user to go to the next step rather than failing the job.
+    if (_pushLocations.isEmpty()) {
+      _logger.error("Push locations cannot be empty. "
+          + "They are needed to get the table config and schema needed for this step. Skipping pre-processing");
+      throw new RuntimeException("Push locations cannot be empty.");
+    }
+    try(ControllerRestApi controllerRestApi = new DefaultControllerRestApi(_pushLocations, _rawTableName)) {
+      _tableConfig = controllerRestApi.getTableConfig();
+      _pinotTableSchema = controllerRestApi.getSchema();
+    }
+
+    if (_tableConfig == null) {
+      _logger.error("Table config cannot be null. Skipping pre-processing");
+      return;
```

Comment:
[Critical] You need to throw exception here, same as line 399.
You might want to use Preconditions.checkState(_tableConfig != null, errorMessage) for better readability